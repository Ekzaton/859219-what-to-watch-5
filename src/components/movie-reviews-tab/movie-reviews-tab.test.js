import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Route, BrowserRouter} from "react-router-dom";

import mockPromo from "../../mocks/promo";
import mockReviews from "../../mocks/reviews";

import MovieReviewsTab from "./movie-reviews-tab";

describe(`MovieReviewsTab snapshot testing`, () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    APP_DATA: {
      movieReviews: mockReviews,
    }
  });

  const movieReviewsTabComponent = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Route>
            <MovieReviewsTab
              movie={mockPromo}
            />
          </Route>
        </BrowserRouter>
      </Provider>
  );

  it(`renders store-connected component correctly`, () => {
    expect(movieReviewsTabComponent.toJSON()).toMatchSnapshot();
  });
});
