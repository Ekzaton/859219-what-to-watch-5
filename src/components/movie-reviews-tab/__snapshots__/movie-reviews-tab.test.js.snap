// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieReviewsTab snapshot testing renders store-connected component correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Christina
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            October 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.3
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie was a tease. The premise was interesting, but they never divulge enough details to make it worthwhile. It ended with far too many questions unanswered.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            John
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            November 24, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Sophie
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            November 4, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        9.6
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          A movie that will take you to another world full of emotions.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Jack
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            November 3, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.6
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Isaac
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            October 29, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        4.6
      </div>
    </div>
  </div>
</div>
`;
